packages/mookme/src/utils/hook-package.ts

```diff
@@ -21,29 +40,78 @@ export function hookPackage(hook: PackageHook): Promise<{ hook: PackageHook; ste
     venvActivate: hook.venvActivate,
   };
 
-  return new Promise((resolve, reject) =>
-    Promise.all(hook.steps.map((step) => runStep(step, options)))
-      .then((errors) => {
-        const hasError = errors.find((err) => err !== null);
-        const result = hasError ? chalk.bgRed.bold(' Error × ') : chalk.bgGreen.bold(' Done ✓ ');
-        loggers[hook.name](`${chalk.bold.inverse(` Hooks : ${hook.name} `)}${result}`);
-        resolve(
-          errors
-            .filter((err) => err !== null)
-            .map((err) => ({
+  const promises = [];
+  const errors: { hook: PackageHook; step: StepCommand; msg: Error }[] = [];
+
+  for (const step of hook.steps) {
+    try {
+      const loaderManager = ui.stepsLoggers[step.name];
+      const stepPromise = runStep(step, options, loaderManager);
+      promises.push(stepPromise);
+
+      if (step.serial) {
+        const stepError = await stepPromise;
```

Comment:
Looks like there's duplicated code here that I think we can consolidate like this:

```
async function handleStep() {
  const stepError = await stepPromise;
  ...
}
if (step.serial) {
  await handleStep();
} else {
  promises.push(handleStep());
}
```

Reply:
Yep that's a good idea, I will do this shortly :)

Reply:
I manage to factorize a bit of this, in function dedicated to handling the step error, however factorizing everything led to weird stuff, like a function dedicated to a step, to which I had to pass the hook, and the ui object for logging purposes. Thus I did not factorize everything.

To my mind, what's remaining will be naturally prettier once we implement a better UI system (with a global instance as most loggers do, instead of creating a ui object at the beginning)